### PR TITLE
Added selection between 1st and 2nd gen CloudSQL for WordPress on App Engine

### PR DESCRIPTION
**Pull request by 1MichaelOHara**
See issue #68 for reference.

### CODE REVIEW

File: appengine/wordpress/src/WordPressSetup.php [hunk 1]
```diff
@@ -57,6 +60,17 @@ protected function configure()
                 null
             )
             ->addOption(
+                'sql_gen',
+                's',
+                InputOption::VALUE_OPTIONAL,
+                'CloudSQL generation to use; 2: '
+                . self::CLOUDSQL_SECOND_GEN
+                . ', 1: '
```
Comment (tmatsuo) [inline appengine/wordpress/src/WordPressSetup.php]: Why using `2` and `1` here and `o` and `t` there? I would also use `f` for First and `s` for Second.

File: appengine/wordpress/src/WordPressSetup.php [hunk 2]
```diff
@@ -74,7 +88,7 @@ protected function configure()
                 'db_instance',
                 null,
                 InputOption::VALUE_OPTIONAL,
-                'Cloud SQL instance id',
+                'Cloud SQL region',
```
Comment (tmatsuo) [inline appengine/wordpress/src/WordPressSetup.php]: Why did you change it to `Cloud SQL region`? Can you add another option instead of changing this?

File: appengine/wordpress/src/WordPressSetup.php [hunk 3]
```diff
@@ -152,7 +166,8 @@ protected function askParameters(
         array &$params,
         InputInterface $input,
         OutputInterface $output,
-        $helper
+        $helper,
+        $sqlGen = false
```
Comment (tmatsuo) [inline appengine/wordpress/src/WordPressSetup.php]: Do you really need to set `db_region` in `askParameters`?

File: appengine/wordpress/src/WordPressSetup.php (hunk 1 above)
Comment (1MichaelOHara) [inline appengine/wordpress/src/WordPressSetup.php]: That is not clean code, I got excited about signing the commit with a GPG key. I was leaning towards using the arguments ```1``` (1st gen) and ```2``` (2nd gen) for the CloudSQL instance generation option. This is because the short version is ```-s``` and the arguments for the environment option already use ```f```  for flexible and ```s``` for standard. @tmatsuo  What are your thoughts?

File: appengine/wordpress/src/WordPressSetup.php (hunk 2 above)
Comment (1MichaelOHara) [inline appengine/wordpress/src/WordPressSetup.php]: Yet again, this is not clean code; got excited about signing the commit with a GPG key. I will revert the string change and add another option for ```db_region```.

File: appengine/wordpress/src/WordPressSetup.php (hunk 3 above)
Comment (1MichaelOHara) [inline appengine/wordpress/src/WordPressSetup.php]: ```$sqlGen``` was need for making a check on the key ```db_region``` to properly concatenate the string with colons for the connection name. In addition, it does not prompt the user for  a region if they are using a 1st gen CloudSQL instance.

File: appengine/wordpress/src/WordPressSetup.php (hunk 1 above)
Comment (tmatsuo) [inline appengine/wordpress/src/WordPressSetup.php]: `1` and `2` work for me

File: appengine/wordpress/src/WordPressSetup.php (hunk 3 above)
Comment (tmatsuo) [inline appengine/wordpress/src/WordPressSetup.php]: I prefer the following flow:

1. If you choose App Engine Flex, just assume to use 2nd gen. If you specify `1`, maybe it can just show error
2. If you choose App Engine Standard, defaults to use 2nd gen. You can override it with specifying an option.
3. If you use 2nd gen, defaults to use `us-central1`, but you can override it with specifying an option
4. Check the value for `db_region` and only allow correct values

Sorry for asking a lot, but I want the script to **always just work**

Thanks for the contribution!

**Comment by tmatsuo** [comment]
@1MichaelOHara Another idea came up to my mind.

There is [an API for administrating Cloud SQL instances](https://cloud.google.com/sql/docs/admin-api/), and [the client library](https://github.com/google/google-api-php-client-services/tree/master/SQLAdmin). With this API, you can list your instances, so that you can create a selection list for users, also you'll know which generation is it, which region it's in, etc in order to construct the connection string. This means the script only need to know the instance name.

Here is the example user experiences:

```
What is your project id: tmatsuo-wp
Detecting available SQL instances...
You have following instances:
1: wp (1st Gen)
2: drupal (2nd Gen)
3: test (1st Gen)
...
...
n: Next page
Which instance to use (1-9, or n for next page): 1
Using Cloud SQL instance "wp".
Downloading WordPress...
...
...
Ready!
```

What do you think? Of course, we can have the intermediate solution with the current approach.

File: appengine/wordpress/src/WordPressSetup.php (hunk 1 above)
Comment (1MichaelOHara) [inline appengine/wordpress/src/WordPressSetup.php]: Sounds good, changed locally.

File: appengine/wordpress/src/WordPressSetup.php (hunk 3 above)
Comment (1MichaelOHara) [inline appengine/wordpress/src/WordPressSetup.php]: All sounds good, just have a few questions. 

1 & 2: My understanding it will prompt users with something like the following. 
```
Please select the App Engine Environment (defaults to Flexible Environment)
  [0] Flexible Environment (Next selection, confirm that it is only CloudSQL 2nd gen instance)
  [1] Standard Environment (Next selection, default to 2nd and offer 1st gen CloudSQL instance) 
```
3: Are you thinking of something like below during the selection

```
Please select the SQL Instance's Region (defaults to us-centeral1)
  [0] us-central1
  [1] us-east1
  [2] ect...
```

4: Just validate the ```db_region``` option's argument against the list of possible regions?

**Comment by 1MichaelOHara** [comment]
@tmatsuo That would be definitely be the optimal solution, but I think that would be out of scope for this pull request. How about making it another issue on GitHub, and I will get this one wrapped up and then we can look into that flow more using CloudSQL admin API?

File: appengine/wordpress/src/WordPressSetup.php (hunk 3 above)
Comment (tmatsuo) [inline appengine/wordpress/src/WordPressSetup.php]: Yes, you are correct.

**Comment by tmatsuo** [comment]
@1MichaelOHara Yes, it sounds good.

**Comment by tmatsuo** [comment]
@1MichaelOHara Do you still want to work on this?

**Comment by tmatsuo** [comment]
@1MichaelOHara ping

**Comment by 1MichaelOHara** [comment]
@tmatsuo My apologies, the project that I was working on moved in another direction. I will not be able to continue working on this pull request.

**Comment by tmatsuo** [comment]
Closing, I opened a new one at #121
